src/Qwiq.Core.Rest/Project.cs

```diff
@@ -64,7 +68,27 @@ internal Project([NotNull] TeamProjectReference project, [NotNull] WorkItemStore
                                               .GetResult();
 
                             return new Node(result).ChildNodes;
-                        }))
+                        }),
+                new Lazy<IQueryFolderCollection>(() =>
+                {
+                    return new QueryFolderCollection(() =>
+                    {
+                        //BUGBUG: There is a bug in the GetQueryAsync in vsts where if a folder contains a '+' character it will return 404, even if the folder exists see here: https://developercommunity.visualstudio.com/content/problem/123660/when-a-saved-query-folder-contains-a-character-it.html
+                        QueryHierarchyItem FolderExpansionFunc(QueryHierarchyItem item)
+                        {
+                            try
+                            {
+                                return store.NativeWorkItemStore.Value.GetQueryAsync(project.Id, item.Path, QueryExpand.Wiql, MaxQueryFolderExpansionDepth).Result;
+                            }
+                            catch(VssServiceResponseException ex){
```

Comment:
For the bug, yes, I'll update the logic to peform the checks and throw a more specific error. Otherwise I don't know we can provide any additional details than the inner exception provides.

Reply:
@pelavall You're right. If you would inspect the error to throw the exception you have (which is super helpful), ensuring what we throw is both helpful and relevant. In other cases, throw the caught exception since we can't add any value